### ISSUE TITLE
About training loss and validation loss.

### ISSUE THREAD

**Issue by Grundwissen** [STVIR/pysot#286]
Hello, that is a really nice tracking project. Good work!

I am playing for a while with this tracking framework.
While training SiamRPN++ with ResNet as backbone, I am plotting the losses in tensorboard. 
I wonder why is there no validation loss? The validation datasets and validation json files are generated and in the right folder. 
Is there any reason to not plot the validation loss? If not, how can I implement it by myself?

Thanks in advance.

**Comment by zyc-ai** [comment]
I'm not sure why they are not added, but you definitely can add the validation loss in `train.py`. 
You may checkout [tensorboard instructions](https://www.tensorflow.org/tensorboard/scalars_and_keras) on how to plot the validation loss

**Comment by ssbagalkar** [comment]
I have the same question. Do we have a solution for this issue? Thanks a lot

**Comment by ssbagalkar** [comment]
@zyc-ai I saw the code inside `pysot/models/model_builder.py`  and I don't think there is any `validation_loss` reported. here is the snippet which only has `total loss`, `cls loss` and `loc_loss`. How can we add validation loss to this? Thanks



    def forward(self, data):

        template = data['template'].cuda()
        search = data['search'].cuda()
        label_cls = data['label_cls'].cuda()
        label_loc = data['label_loc'].cuda()
        label_loc_weight = data['label_loc_weight'].cuda()

        # get feature
        zf = self.backbone(template)
        xf = self.backbone(search)
        if cfg.MASK.MASK:
            zf = zf[-1]
            self.xf_refine = xf[:-1]
            xf = xf[-1]
        if cfg.ADJUST.ADJUST:
            zf = self.neck(zf)
            xf = self.neck(xf)
        cls, loc = self.rpn_head(zf, xf)

        # get loss
        cls = self.log_softmax(cls)
        cls_loss = select_cross_entropy_loss(cls, label_cls)
        loc_loss = weight_l1_loss(loc, label_loc, label_loc_weight)

        outputs = {}
        outputs['total_loss'] = cfg.TRAIN.CLS_WEIGHT * cls_loss + \
            cfg.TRAIN.LOC_WEIGHT * loc_loss
        outputs['cls_loss'] = cls_loss
        outputs['loc_loss'] = loc_loss

        if cfg.MASK.MASK:
            # TODO
            mask, self.mask_corr_feature = self.mask_head(zf, xf)
            mask_loss = None
            outputs['total_loss'] += cfg.TRAIN.MASK_WEIGHT * mask_loss
            outputs['mask_loss'] = mask_loss
        return outputs

**Comment by zyc-ai** [comment]
> @zyc-ai I saw the code inside `pysot/models/model_builder.py` and I don't think there is any `validation_loss` reported. here is the snippet which only has `total loss`, `cls loss` and `loc_loss`. 

If you are training, it's training loss, if not, it's validation loss.
